src/nvim/os/time.c

```diff
@@ -144,6 +144,44 @@ struct tm *os_localtime(struct tm *result) FUNC_ATTR_NONNULL_ALL
   return os_localtime_r(&rawtime, result);
 }
 
+/// Portable version of POSIX ctime_r()
+///
+/// @param clock[in]
+/// @param result[out] Pointer to a 'char' where the result should be placed
+/// @return human-readable string of current local time
+char *os_ctime_r(const time_t *restrict clock, char *restrict result)
```

Comment:
This and `os_ctime` are copied from `os_localtime_r` and `os_localtime`. Main difference is that they don't return `NULL` and they use `char *` for result output.

Why does Vim not use `ctime_r`?

Reply:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=47802#c14
https://gcc.gnu.org/legacy-ml/gcc-patches/2014-05/msg02124.html
https://gcc.gnu.org/legacy-ml/gcc-patches/2014-05/txt7ZCnLNNUaw.txt

Suggestion is to use `localtime_r` and `snprintf`, not strftime, for portability and MINGW workaround.